src/sonic-config-engine/portconfig.py

```diff
@@ -423,9 +426,10 @@ def parse_platform_json_file(hwsku_json_file, platform_json_file):
 
         # take optional fields from hwsku.json
         for child_port in child_ports:
-            for key, item in hwsku_dict[INTF_KEY][child_port].items():
-                if key in OPTIONAL_HWSKU_ATTRIBUTES:
-                    child_ports.get(child_port)[key] = item
+            if child_port in hwsku_dict[INTF_KEY]:
```

Comment:
better to assign a variable for `hwsku_dict[INTF_KEY]` to avoid looking up it in every iteration